### DISCUSSION TITLE
Single action buttons

### DISCUSSION THREAD

**Discussion by dietercoopman** [rappasoft/laravel-livewire-tables#232] [Feature Requests]
Hi,

i'm switching to the new release of your package. Nice work 🤩🤩

In a pull request on the "old" repo I created the functionality to add an "add record" button ... For example an Add button in the header off the table.

It's in fact very strange none of the datatables i've tested have this capability ... So my question is , how do you implement an "add record" button in real life applications ? 

Could be implemented with a "actions" array on class thats shows buttons.

Now i've fixed it like this , but thats not so nice in the gui 

![Schermafbeelding 2021-04-23 om 15 50 59](https://user-images.githubusercontent.com/4672752/115881671-452d5280-a44c-11eb-8f68-cfbd1accf09e.png)

What is your idea on this ? 

Greetings Dieter

**Comment by rappasoft** [comment]
Since my tables are only one component on my page, I usually do it outside of it. I have a parent livewire component that controls the page, the table lives in it. The "Create/New" button lives on the parent component and usually opens a modal and the create functionality lives on the parent livewire component and then just refreshes the table when created.

**Reply by rappasoft** [reply]
Like this:

<img width="1533" alt="Screen Shot 2021-04-23 at 10 04 14" src="https://user-images.githubusercontent.com/6026504/115883022-5611a000-a41b-11eb-9f7b-0ca818113b73.png">
<img width="1538" alt="Screen Shot 2021-04-23 at 10 03 53" src="https://user-images.githubusercontent.com/6026504/115883024-5611a000-a41b-11eb-854e-5f8146d77429.png">

**Reply by dietercoopman** [reply]
Great , i'll do it that way ... thx !

**Reply by rappasoft** [reply]
No problem!

**Reply by dietercoopman** [reply]
Ok i can live with it 🤩 🤩 , i've implemented like this

![Schermafbeelding 2021-04-23 om 16 21 19](https://user-images.githubusercontent.com/4672752/115885196-00a3b600-a450-11eb-8789-7646bb99ed56.png)

**Reply by AEK-BKF** [reply]
@rappasoft Could you please share a code example for this ? Thanks

**Reply by rappasoft** [reply]
@AEK-BKF coding example for what?

**Reply by AEK-BKF** [reply]
How handle the CRUD following this approach parent child.

**Reply by rappasoft** [reply]
This really has nothing to do with this package. You can handle CRUD anyway you want. In the above example, he has a button outside the table that probably opens a modal. The modal can create the resource and either refresh the table or redirect somewhere else. You can have an actions column on the table with edit/delete buttons that also open other modals or redirect somewhere.

The table is just a container for the rows in your database to display how you want, put it in a parent livewire component and use that components methods to control your CRUD.

**Reply by AEK-BKF** [reply]
Thank you :)

**Comment by rappasoft** [comment]
It honestly makes sense to me this way, you have full control over it now. Maybe I can add an action buttons area in a future release.

**Reply by dietercoopman** [reply]
Nah , you're 100% right ... looks good like that.  No need for the actions area in my opinion.  It adds too much complexity to your package that needs to be in control of the enduser.  I totally agree
